### PR TITLE
Update auth demo to include password validation

### PR DESCRIPTION
**Pull request by ch5zzy**
Update the auth demo to include a section for password validation. As a password is typed in, the status of requirements will be displayed and updated based on what is required by the policy.

See a quick demo at https://drive.google.com/file/d/1WlAkDJrZCS5qiMYSrNLk-LDWjquFPxTA/view?usp=sharing

### CODE REVIEW

**Review by renkelvin — APPROVED** [review]

File: packages/auth/demo/src/index.js [hunk 1]
```diff
@@ -493,6 +494,106 @@ function onInitializeRecaptchaConfig() {
...
+        status.containsUppercaseLetter
+      );
+      setRequirementStatus(
+        '#password-validation-contains-numeric',
+        status.containsNumericCharacter
+      );
+      setRequirementStatus(
+        '#password-validation-contains-non-alphanumeric',
+        status.containsNonAlphanumericCharacter
+      );
```
Comment (Xiaoshouzi-gh) [inline packages/auth/demo/src/index.js:569]: can you use a for loop here to iterate thru the keys using `Object.keys(passwordPolicy)`.  Using the property as ID is acceptable to me, but you can also manipulate the string if you want with a password-validation prefix.

**Review by Xiaoshouzi-gh — COMMENTED** [review]
Will there be separate PRs to add the password policy check on existing sign in, sign up & reset password flow?

**Comment by ch5zzy** [comment]
> Will there be separate PRs to add the password policy check on existing sign in, sign up & reset password flow?

Since password validation is not being implemented as part of these flows, I figured it was best to have a separate section for testing password validation. We could have a button toggle for each of these flows (like enable/disable password validation) with a similar UI to this for displaying the requirements, but I wasn't sure if we wanted to duplicate this UI across these sections or just have one section for demoing password validation.

File: packages/auth/demo/src/index.js (hunk 1 above)
Comment (ch5zzy) [reply packages/auth/demo/src/index.js:569]: Is it okay that the IDs would then be in camelCase instead of kebab-case like the other IDs?

Comment (Xiaoshouzi-gh) [reply packages/auth/demo/src/index.js:569]: you can try use a regex replace/a helper method to format them to kabob like strings.

Comment (ch5zzy) [reply packages/auth/demo/src/index.js]: Done.

**Review by Xiaoshouzi-gh — APPROVED** [review]

**Pull request merged**